packages/core/getUserAvailability.ts

```diff
@@ -413,9 +416,57 @@ const _getUserAvailability = async function getUsersWorkingHoursLifeTheUniverseA
     end: dayjs(busy.end),
   }));
 
-  const dateRangesInWhichUserIsAvailable = subtract(dateRanges, formattedBusyTimes);
+  let dateRangesInWhichUserIsAvailable = subtract(dateRanges, formattedBusyTimes);
+
+  let dateRangesInWhichUserIsAvailableWithoutOOO = subtract(oooExcludedDateRanges, formattedBusyTimes);
```

Comment:
I think we need an early return here, and it doesn't make sense to perform other operations if the length is zero

Reply:
Thanks @anikdhabal! Yes correct!

But instead of early return added a check, to execute the whole first-slot logic block only if `dateRangesInWhichUserIsAvailableWithoutOOO` is not empty, just to preserve the debug logs done below.

Reply:
Yeah, I mean the same thing.